src/libcore/alloc/layout.rs

```diff
@@ -258,19 +258,17 @@ impl Layout {
 
     /// Creates a layout describing the record for `self` followed by
     /// `next`, including any necessary padding to ensure that `next`
-    /// will be properly aligned. Note that the resulting layout will
-    /// satisfy the alignment properties of both `self` and `next`.
-    ///
-    /// The resulting layout will be the same as that of a C struct containing
-    /// two fields with the layouts of `self` and `next`, in that order.
+    /// will be properly aligned, but no trailing padding. Note that
+    /// the resulting layout will satisfy the alignment properties of
+    /// both `self` and `next`, in order to ensure field alignment.
```

Comment:
Thanks, this is already a lot better than before.

I wonder if we could do with a bit more emphasis. For once, "no trailing padding" could become `*no trailing padding*` to make it stand out a bit more. Second, I think this should mention explicitly that almost any time you use `extend`, you want to later use `pad_to_align`. Maybe it should have example code (or link to example code) [like this](https://doc.rust-lang.org/nightly/reference/type-layout.html#reprc-structs) but using `Layout` instead.

Reply:
I emphasized *no trailing padding* as suggested.

[The real `Layout`-based code for `repr(C)` layout](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=424fa86a7f3c9a6d920fbbe056ac2c48) necessarily uses const generics; it's not very useful if it doesn't, as you could just use `Layout::new`. I'm in favor of adding this method to `Layout` eventually.

I'm not sure how best to stick a reminder to call `pad_to_align` here. Perhaps

```rust
    /// will be properly aligned, but with no trailing padding. You probably
    /// need to call `pad_to_align` to add the trailing padding before this
    /// `Layout` represents a layout compatible with an actual allocation.
    ///
    /// Note that the resulting layout will satisfy the alignment properties
    /// of both `self` and `next`, in order to ensure alignment of both parts.
```

?

Reply:
> The real Layout-based code for repr(C) layout necessarily uses const generics; it's not very useful if it doesn't, as you could just use Layout::new. I'm in favor of adding this method to Layout eventually.

I am aware. I did not mean a fully generic method, but example code -- e.g., it could return a `Vec<usize>` for the offsets, and take `fields: &[Layout]`.

Reply:
> I'm not sure how best to stick a reminder to call pad_to_align here. Perhaps

I'd make it slightly stronger, like "You should call `pad_to_align` after adding all fields", or so.

Reply:
[I added wording along those lines, complete with example.](https://github.com/rust-lang/rust/pull/69362/commits/65d450552fb8ac503d557f839dec2a77b6b67021) I'm not _perfectly_ happy with the wording, but it should be good enough.

Reply:
Thanks, this looks good overall, I just have some nits.